code/modules/xenoarchaeology/tools/console/xenoarchaeology_console.dm

```diff
@@ -0,0 +1,261 @@
+//TODO: Fix a million artifacts spawning in one crate - Racc
+///Stability lost on purchase
+#define STABILITY_COST 30
+///Stability gained on-tick
+#define STABILITY_GAIN 5
+
+/obj/machinery/computer/xenoarchaeology_console
+	name = "research and development listing console"
+	desc = "A science console used to source sellers, and buyers, for various blacklisted research objects."
+	icon_screen = "xenoartifact_console"
+	icon_keyboard = "rd_key"
+	circuit = /obj/item/circuitboard/computer/xenoarchaeology_console
+
+	///Which science server receives points
+	var/datum/techweb/linked_techweb
+	///Which department's budget receives profit
+	var/datum/bank_account/budget
+
+	///List of current listing sellers
+	var/list/sellers = list(/datum/rnd_lister/artifact_seller/bastard, /datum/rnd_lister/artifact_seller/uranium_bananium,
+	/datum/rnd_lister/artifact_seller/bluespace, /datum/rnd_lister/artifact_seller/plasma_bluespace)
+
+	///radio used by the console to send messages on science channel
+	var/obj/item/radio/headset/radio
+	///Do we do purchase notices on the radio?
+	var/radio_purchase_notice = TRUE
+	///Do we do solved notices on the radio?
+	var/radio_solved_notice = TRUE
+
+	///List of active orders
+	var/list/console_orders = list()
+	///Max contents per order - leave this as a variable, trust
+	var/max_pack_contents = 5
+
+	///History of purchases and sales
+	var/list/history = list()
+
+	///Is this console the main character?
+	var/main_console = FALSE
+
+/obj/machinery/computer/xenoarchaeology_console/Initialize()
+	. = ..()
+	//Link up with SS to see if we're the choosen one
+	if(!SSxenoarchaeology.main_console)
+		SSxenoarchaeology.register_console(src)
+	RegisterSignal(SSxenoarchaeology, XENOA_NEW_CONSOLE, PROC_REF(be_the_guy))
+	//Link relevant stuff
+	linked_techweb = SSresearch.science_tech
+	budget = SSeconomy.get_budget_account(ACCOUNT_SCI_ID)
+	var/list/new_sellers = sellers.Copy()
```

Comment:
I am not certain for copying this seller list... not sure. I just wanted to leave my concern rather than leaving no mark - nevermind this.